packages/project-access/src/file/file-search.ts

```diff
@@ -136,3 +137,18 @@ export async function findFileUp(fileName: string, startPath: string, fs?: Edito
         return dirname(startPath) !== startPath ? findFileUp(fileName, dirname(startPath), fs) : undefined;
     }
 }
+
+/**
+ * @description Returns a flat list of all file paths under a directory tree,
+ * recursing through all subdirectories
+ * @param {string} dir - the directory to walk
+ * @returns {string[]} - array of file path strings
+ * @throws if an error occurs reading a file path
+ */
+export function getFilePaths(dir: string): string[] | [] {
+    return readdirSync(dir).reduce((files: string[], entry: string) => {
+        const entryPath = join(dir, entry);
+        const isDirectory = statSync(entryPath).isDirectory();
+        return isDirectory ? [...files, ...getFilePaths(entryPath)] : [...files, entryPath];
```

Comment:
Hi @cianmSAP, can we please use asynchronous functions for file access to keep it in alignment with the rest of this module.

Reply:
Updated. It resulted in a lot of updates needed in packages/ui5-application-writer/src/options.ts to await the result...